src/mono/mono/utils/mono-wasm-pagemgr.h

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// Provides a simple, efficient implementation of anonymous mmap/munmap for WASM.
+// The emscripten libc version of mmap for anonymous mappings is slow during startup
```

Comment:
Is this candidate for upstream patch ?

Reply:
I have an upstream issue open, but it's not easy to fix this there.

Reply:
https://github.com/emscripten-core/emscripten/issues/21620